core/classes/form.php

```diff
@@ -777,3 +811,44 @@ public function parse(SpoonTemplate $template = null)
         return $output;
     }
 }
+
+/**
+ * @author Wouter Sioen <wouter@sumocoders.be>
+ *
+ * A Custom textfield containing comma seperated tags
+ */
+class SiteFormTags extends SpoonFormText
+{
+    /**
+     * Note that the variable name is only allowHTML to be compliant with the parent.
+     * It should be named "returnObjects" or something similar.
+     *
+     * @param bool $allowHTML Will objects be returned?
+     */
+    public function getValue($allowHTML = null)
+    {
+        if ($allowHTML === true) {
+            $tags = explode('|', parent::getValue());
+
+            // get the existing tag or fallback to a new one
+            foreach ($tags as $key => $tag) {
+                if (empty($tag)) {
+                    unset($tags[$key]);
+                }
+
+                try {
+                    $tags[$key] = Tag::getByName($tag);
+                } catch (\InvalidArgumentException $ex) {
```

Comment:
- You're currently not in a namespace, so the fully qualified classname should not be used (This prevents you from refactoring using namespace/class aliases in use statements later on)
- i would use `$e or `$exception` instead of `$ex`
- It would be great if a dedicated exception would be thrown, like TagNotFoundException.